README.md

```diff
@@ -145,37 +132,116 @@ Here is an example:
 
-If the `vp_token` or `vc_token` is returned in the frontchannel, a hash of the respective token MUST be included in `id_token` (see next chapter).
+This specification uses the `_claim_names_` element as is. It is used to map claims as requested by the RP to the respective source in this case a verifiable credential or presentation. 
 
-## ID Token Extensions
+This specification introduces a new top level claim `_credential_types`, which maps credential types as requested by the RP to  source used to provide the respective credential. 
 
-This section defines the extensions to the ID Token structure used or defined in this specification. 
+This specification extends the syntax for aggregated and disctributed claims sources as follows:
 
-`vc` - see https://www.w3.org/TR/vc-data-model/#json-web-token-extensions
+Aggregated and distributed claims sources:
 
+* a `format` field is added containing the format of the verifiable credential or presentation. `vp_jwt` and `vc_jwt` denote a credential or presentation in JWT format, respectively. `vc_ldp` and `vp_ldp` denote a credential or presentation in JSON-LD, respectively. All content is provided in the claim source object as is without further encoding. For backward compability, the format `jwt` is used to denote standard OIDC aggregated/distributed claims. 
+* a `value` element containing the actual presentation or credential in case of an aggregated claim for verifiable presentations or credentials. 
 
-`vp` - see https://www.w3.org/TR/vc-data-model/#json-web-token-extensions
-
-`vp_hash`
-OPTIONAL. Hash value of `vp_token` that represents the W3C VP. Its value is the base64url encoding of the left-most half of the hash of the octets of the ASCII representation of the `vp_token` value, where the hash algorithm used is the hash algorithm used in the alg Header Parameter of the ID Token's JOSE Header. For instance, if the alg is RS256, hash the vp_token value with SHA-256, then take the left-most 128 bits and base64url encode them. The `vp_hash` value is a case sensitive string.
-
-`vc_hash`
-OPTIONAL. Hash value of `vc_token` that represents the W3C VC. Its value is the base64url encoding of the left-most half of the hash of the octets of the ASCII representation of the `vc_token` value, where the hash algorithm used is the hash algorithm used in the alg Header Parameter of the ID Token's JOSE Header. For instance, if the alg is RS256, hash the `vc_token` value with SHA-256, then take the left-most 128 bits and base64url encode them. The `vc_hash` value is a case sensitive string.
+Here is an example:
 
+```json
+{
+   "iss":"https://self-issued.me",
+   "sub":"248289761001",
+   "_claim_names":{
+      "family_name":"src1",
+      "alumniof":"src2",
+      "nationality":"src3",
+      "data_of_birth":"src4"
+   },
+   "_credential_types":{
+      "https://www.w3.org/2018/credentials/examples/v1/AlumniCredential":[
+         "src1",
+         "src2"
+      ],
```

Comment:
Is it common that a claim name / credential type would be mapped to multiple sources?
I am struggling to think of a situation for this...
How would the RP determine how to proceed?

Reply:
what do you think is reasonable? one source, multiple credentials (VP) but one credential at most one source?